include/dlaf/matrix/internal/tile_future_manager.h

```diff
@@ -0,0 +1,85 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#include <hpx/hpx.hpp>
+#include "dlaf/matrix/tile.h"
+
+namespace dlaf {
+namespace matrix {
+namespace internal {
+
+template <class ReturnTileType, class TileType>
+hpx::future<ReturnTileType> setPromiseTileFuture(hpx::future<TileType>& old_future,
+                                                 hpx::lcos::local::promise<TileType>& p) noexcept {
+  return old_future.then(hpx::launch::sync, [p = std::move(p)](hpx::future<TileType>&& fut) mutable {
```

Comment:
`std::move(old_future).then(...)`

Reply:
I don't understand. Why should I move old_future to call its member?

On the other hand if you mean `std::move(old_future.then(...))` `.then` already returns an rvalue.

Reply:
No, I meant what I wrote but it's OK either way. If you accept the parameter by value, by calling std::move, the compiler may produce better codegen as that would indicate that the data stored in the variable is not used later in the function. The function is small, so the compiler may already see that.